src/components/select/multi-select/multi-select.component.js

```diff
@@ -326,11 +327,14 @@ const MultiSelect = React.forwardRef(
       };
     }, [handleGlobalClick]);
 
+    const onFilterChange = useStableCallback(onFilterChangeProp);
+    const isFirstRender = useRef(true);
     useEffect(() => {
-      if (onFilterChange) {
+      if (onFilterChange && !isFirstRender.current) {
         onFilterChange(filterText);
       }
-    }, [filterText, onFilterChange]);
+      isFirstRender.current = false;
```

Comment:
as in filterable-select, if we don't extract this to a custom hook I would much prefer this update is in its own `useEffect`